api/types/saml.go

```diff
@@ -724,7 +569,7 @@ func (*teleportSAMLConnectorMarshaler) UnmarshalSAMLConnector(bytes []byte, opts
 			}
 		}
 
-		if err := c.Metadata.CheckAndSetDefaults(); err != nil {
```

Comment:
This seemed like a mistake, or `CheckAndSetDefaults` caused issues in its previous state.

Reply:
What kind of issues?

Reply:
Basically the logic I moved out of `CheckAndSetDefaults` was doing more than the basic checks it should have. So most likely `c.Metadata.CheckAndSetDefaults` was called to avoid calling this extra logic. In any case, `c.CheckAndSetDefaults` should be called here instead now.